modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -249,8 +250,8 @@ public Object result(@PathParam("executionId") String executionId) throws IOExce
             // cleanup after result is accessed
             cleanup(executionId);
 
-            if (status != 0) {
-                throw new NuxeoException(error, status);
+            if (status != 0 || StringUtils.isNotEmpty(error)) {
```

Comment:
Do we have legitimate cases where `error` can be just `""`? If not, I'd prefer a simple `null` check. Same in `getError`